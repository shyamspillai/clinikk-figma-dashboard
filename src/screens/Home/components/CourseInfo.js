import { Avatar, Button } from 'antd'
import { ClockCircleFilled, FireFilled } from '@ant-design/icons'

import styled from 'styled-components'

const CourseInfo = ({ title, author, rating, time, icon }) => {

    return (
        <CourseInfoBar>
            <StyledAvatar shape="square" size={50} icon={icon} />
            <div className="flex column" style={{ width: "100px" }}>
                <StyledCourseTitle
                    data-test="course-title"
                    className="baloo size-16 weight-600">
                    {title}
                </StyledCourseTitle>
                <StyledCourseAuthor
                    data-test="course-author"
                    className="roboto size-13 weight-400">
                    by {author}
                </StyledCourseAuthor>
            </div>

            <TimeIconHolder>
                <ClockCircleFilled /><span data-test="course-time"> {time}</span>
            </TimeIconHolder>

            <RatingIconHolder>
                <FireFilled /> <span data-test="course-rating">{rating}</span>
            </RatingIconHolder>

            <Button
                style={{
                    border: '1px solid #000000',
                    borderRadius: "10px",
                    color: "#000000"
                }}
                className="roboto size-13 weight-700"
            >
                Continue
            </Button>
        </CourseInfoBar>
    )
}

const StyledAvatar = styled(Avatar)`
    @media only screen and (max-width: 600px) {
        height: 30;
        width: 30;
    }
`;

const CourseInfoBar = styled.div`
    width: 100%;
    background-color: #f5f5f7;
    height: 75px;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    align-items: center;
    border-radius: 15px;
    margin-top: 20px;

    @media only screen and (max-width: 600px) {
        flex-wrap: wrap;
        width: 300px;
    }
`;

const StyledCourseAuthor = styled.span`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
`;

const StyledCourseTitle = styled.span`
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
`;

const TimeIconHolder = styled.div`
    @media only screen and (max-width: 600px) {
        display: none;
    }
`

const RatingIconHolder = styled.div`
    @media only screen and (max-width: 600px) {
        display: none;
    }
`

export default CourseInfo
