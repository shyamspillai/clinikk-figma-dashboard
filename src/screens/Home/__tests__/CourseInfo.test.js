import React from 'react';
import CourseInfo from '../components/CourseInfo';

import Enzyme, { render, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

describe('Course Info component', () => {

  const props = {
    title: 'Sample title',
    author: 'Sample author',
    rating: 'Sample rating',
    time: 'Sample time'
  }

  it('should match snapshot', () => {
    const CourseInfoComponent = shallow(<CourseInfo />);
    expect(toJson(CourseInfoComponent)).toMatchSnapshot();
  })

  it('should render the course stat props correctly', () => {
    const wrapper = render(
      <CourseInfo {...props} />
    );

    expect(wrapper.find("[data-test='course-title']").text()).toBe(props.title);
    expect(wrapper.find("[data-test='course-author']").text()).toBe("by " + props.author);
    expect(wrapper.find("[data-test='course-rating']").text()).toBe(props.rating);
    expect(wrapper.find("[data-test='course-time']").text()).toBe(" " + props.time);

  });

});