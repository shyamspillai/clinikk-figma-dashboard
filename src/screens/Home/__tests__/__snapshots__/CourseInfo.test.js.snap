// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Info component should match snapshot 1`] = `
<styled.div>
  <Styled(Avatar)
    shape="square"
    size={50}
  />
  <div
    className="flex column"
    style={
      Object {
        "width": "100px",
      }
    }
  >
    <styled.span
      className="baloo size-16 weight-600"
      data-test="course-title"
    />
    <styled.span
      className="roboto size-13 weight-400"
      data-test="course-author"
    >
      by 
    </styled.span>
  </div>
  <styled.div>
    <ForwardRef(ClockCircleFilled) />
    <span
      data-test="course-time"
    >
       
    </span>
  </styled.div>
  <styled.div>
    <ForwardRef(FireFilled) />
     
    <span
      data-test="course-rating"
    />
  </styled.div>
  <Button
    className="roboto size-13 weight-700"
    style={
      Object {
        "border": "1px solid #000000",
        "borderRadius": "10px",
        "color": "#000000",
      }
    }
  >
    Continue
  </Button>
</styled.div>
`;
